pkg/virt-handler/node-labeller/node_labeller.go

```diff
@@ -0,0 +1,269 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package nodelabeller
+
+import (
+	"context"
+	"encoding/json"
+	"reflect"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/workqueue"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/client-go/log"
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+	device_manager "kubevirt.io/kubevirt/pkg/virt-handler/device-manager"
+)
+
+const (
+	DeprecatedLabelNamespace              = "feature.node.kubernetes.io"
+	DeprecatedLabellerNamespaceAnnotation = "node-labeller-feature.node.kubernetes.io"
+	kvmPath                               = "/dev/kvm"
+	DeprecatedcpuFeaturePrefix            = "/cpu-feature-"
+	DeprecatedcpuModelPrefix              = "/cpu-model-"
+	DeprecatedHyperPrefix                 = "/kvm-info-cap-hyperv-"
+	CPUFeaturePrefix                      = "cpu-feature.node.kubevirt.io/"
+	cpuModelPrefix                        = "cpu-model.node.kubevirt.io/"
+	hypervPrefix                          = "hyperv.node.kubevirt.io/"
+)
+
+//NodeLabeller struct holds informations needed to run node-labeller
+type NodeLabeller struct {
+	kvmController     *device_manager.DeviceController
+	clientset         kubecli.KubevirtClient
+	host              string
+	namespace         string
+	logger            *log.FilteredLogger
+	clusterConfig     *virtconfig.ClusterConfig
+	hypervFeatures    supportedFeatures
+	hostCapabilities  supportedFeatures
+	queue             workqueue.RateLimitingInterface
+	supportedFeatures []string
+	cpuInfo           cpuInfo
+}
+
+func NewNodeLabeller(kvmController *device_manager.DeviceController, clusterConfig *virtconfig.ClusterConfig, clientset kubecli.KubevirtClient, host, namespace string) *NodeLabeller {
+	return &NodeLabeller{
+		kvmController: kvmController,
+		clientset:     clientset,
+		host:          host,
+		namespace:     namespace,
+		logger:        log.DefaultLogger(),
+		clusterConfig: clusterConfig,
+		queue:         workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
+	}
+}
+
+//Run runs node-labeller
+func (n *NodeLabeller) Run(threadiness int, stop chan struct{}) {
+	defer n.queue.ShutDown()
+	n.logger.Infof("node-labeller is running")
+
+	err := n.loadCPUInfo()
+	if err != nil {
+		n.logger.Errorf("node-labeller could not load cpu info: " + err.Error())
+		return
+	}
+
+	err = n.loadHostSupportedFeatures()
+	if err != nil {
+		n.logger.Errorf("node-labeller could not load supported features: " + err.Error())
+		return
+	}
+
+	err = n.loadHostCapabilities()
+	if err != nil {
+		n.logger.Errorf("node-labeller could not load host capabilities: " + err.Error())
+		return
+	}
```

Comment:
It looks like these errors should cause virt-handler to fail. I'd suggest moving this code block to the "NewNodeLabeller()" function and having that function return an error.

Then in the cmd/virt-handler/virt-handler.go code, i'd expect the setup to panic if NewNodeLabeller returns an error. 

```
        err = NewNodeLabeler(<args> ) 
	if err != nil {
		panic(err)
	}
```